lib/ansible/plugins/connection/kubernetes.py

```diff
@@ -0,0 +1,337 @@
+# Based on the local connection plugin and methods from the Docker plugin
+#
+# Connection plugin for kubernetes pods
+# (c) 2017, Joseph Callen <jcallen@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+'''
+DOCUMENTATION:
+    connection: kubernetes
+    short_description: connect via kubernetes api to a pod
+    description:
+        - This connection plugin allows ansible to communicate to the target pod via kubernetes python client
+    author: Joseph Callen
+    version_added: 2.4
+    options:
+        _host:
+            description: Hostname/ip to connect to.
+            default: inventory_hostname
+            host_vars:
+                 - ansible_host
+                 - ansible_ssh_host
+        _password:
+            description: Authentication password for the C(remote_user). Can be supplied as CLI option.
+            host_vars:
+                - ansible_password
+                - ansible_ssh_pass
+        kubernetes_namespace:
+            description: The kubernetes namespace where the pods are located
+            host_vars:
+                - ansible_kubernetes_namespace
+        kubernetes_cluster:
+            description:
+                - The ip address or DNS name of the kubernetes cluster
+            host_vars:
+                - ansible_kubernetes_cluster
+        kubernetes_config_file
+            description:
+                - The path and filename of the kubernetes config file that will be used when connecting
+                - to the kubernetes cluster
+            host_vars:
+                - ansible_kubernetes_config_file
+        port:
+            description: Remote port to connect to.
+            type: int
+            config:
+               - section: defaults
+                 key: remote_port
+                 default: 22
+            env_vars:
+               - ANSIBLE_REMOTE_PORT
+            host_vars:
+               - ansible_port
+               - ansible_ssh_port
```

Comment:
ssh is for ssh/paramiko connection plugins, you might want to use kubernetes here